lib/sles4sap/sdaf_library.pm

```diff
@@ -0,0 +1,519 @@
+
+use strict;
+use warnings;
+use testapi;
+use Exporter qw(import);
+use Carp qw(croak);
+use mmapi qw(get_current_job_id);
+use utils qw(write_sut_file);
+use File::Basename;
+use Regexp::Common qw(net);
+
+=head1 SYNOPSIS
+
+Library with common functions for Microsoft SDAF deployment automation. Documentation can be found on the
+L<projects official website|https://learn.microsoft.com/en-us/azure/sap/automation/get-started>
+
+Github repositories:
+L<Automation scripts|https://github.com/Azure/sap-automation/tree/main>
+L<Sample configurations|https://github.com/Azure/SAP-automation-samples/tree/main>
+
+Basic terminology:
+
+B<SDAF>: SAP deployment automation framework
+
+B<Control plane>: Common term for Resource groups B<Deployer> and B<Library>.
+Generally it is part of a permanent infrastructure in the cloud.
+
+B<Deployer>: Resource group providing services such as keyvault, Deployer VM and associated resources.
+
+B<Deployer VM>: Central point that contains SDAF installation and where the deployment is executed from.
+Since SUT VMs have no public IPs, this is also serving as a jumphost to reach them via SSH.
+
+B<Library>: Resource group providing storage for terraform state files, SAP media and private DNS zone.
+
+B<Workload zone>: Resource group that provides services similar to support server.
+
+B<SAP Systems>: Resource group containing SAP SUTs and related resources.
+
+=cut
+
+our @EXPORT = qw(
+  az_login
+  sdaf_prepare_ssh_keys
+  sdaf_get_deployer_ip
+  serial_console_diag_banner
+  set_common_sdaf_os_env
+  prepare_sdaf_repo
+  cleanup_sdaf_files
+);
+
+
+=head2 homedir
+
+    homedir();
+
+Returns home directory path for current user from env variable $HOME.
+
+=cut
+
+sub homedir {
+    return (script_output('echo $HOME'));
+}
+
+=head2 deployment_dir
+
+    deployment_dir([, create=>$create]);
+
+B<create>: Create directory if it does nto exist.
+
+Returns deployment directory path with job ID appended as unique identifier.
+Optionally it can create directory if it does nto exists.
+
+=cut
+
+sub deployment_dir {
+    my (%args) = @_;
+    my $deployment_dir = get_var('DEPLOYMENT_ROOT_DIR', '/tmp') . '/Azure_SAP_Automated_Deployment_' . get_current_job_id();
+    assert_script_run("mkdir -p $deployment_dir") if $args{create};
+    return $deployment_dir;
+}
+
+=head2 log_dir
+
+    log_dir([, create=>$create]);
+
+B<create>: Create directory if it does nto exist.
+
+Returns logging directory path with job ID appended as unique identifier.
+Optionally creates the directory.
+
+=cut
+
+sub log_dir {
+    my (%args) = @_;
+    my $log_dir = deployment_dir() . '/openqa_logs';
+    assert_script_run("mkdir -p $log_dir") if $args{create};
+    return $log_dir;
+}
+
+=head2 variable_file
+
+    variable_file();
+
+Returns full path to a file containing all required SDAF OS env variables.
+Sourcing this file is essential for running SDAF.
+
+=cut
+
+sub variable_file {
+    return deployment_dir() . '/sdaf_variables';
```

Comment:
who provide `sdaf_variables` ? Does it make sense to mention it in the doc part? Seems it is `create_sdaf_os_var_file`.

Reply:
This is just a function to get the path for variable file. 
Variables are provided by exported function `set_common_sdaf_os_env`  it already has a link to the official doc with variable descriptions.
`create_sdaf_os_var_file` is private function but I'll add the link there as well.